src/views/List.jsx

```diff
@@ -1,15 +1,69 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	// state for input
+	const [input, setInput] = useState('');
+	const [searchData, setSearchData] = useState(data);
+	const [isValid, setIsValid] = useState(false);
+	const [searchLength, setSearchLength] = useState(1);
+	// Handle event
+	const handleInputChange = (e) => {
+		setInput(e.target.value);
+	};
+
+	const handleInputClear = () => {
+		setInput('');
+	};
+
+	useEffect(() => {
+		const searchRegex = new RegExp(input, 'i');
```

Comment:
Consider changing this regex to:
`const searchRegex = new RegExp(input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'i');`

Right now if the user types in any of the special regex characters as their first character the whole page breaks. (for example *)